src/Workspaces/MSBuildTest/Utilities/DotNetSdkMSBuildInstalled.cs

```diff
@@ -111,22 +33,29 @@ static string GetSolutionFolder()
                 return solutionFolder;
             }
         }
-#endif
 
-        public DotNetSdkMSBuildInstalled()
+        private static bool HasNetCoreSdkForSolution(string solution)
         {
+            BuildHostProcessManager? buildHostProcessManager = null;
+
+            try
+            {
+                buildHostProcessManager = new BuildHostProcessManager();
+
+                var buildHost = buildHostProcessManager.GetBuildHostAsync(BuildHostProcessManager.BuildHostProcessKind.NetCore, CancellationToken.None).Result;
+
+                return buildHost.HasUsableMSBuildAsync(solution, CancellationToken.None).Result;
+            }
+            finally
+            {
+                buildHostProcessManager?.DisposeAsync().AsTask().Wait();
+            }
         }
 
-        // Eventually should revert condition to 'SdkPath is null' after MSBuild issue is fixed:
-        // https://github.com/dotnet/roslyn/issues/67566
         public override bool ShouldSkip
-            => true;
+            => s_skipReason is not null;
 
         public override string SkipReason
-#if NETCOREAPP
-            => $"Could not locate .NET SDK version {VersionString}.";
-#else
-            => $"Test runs on .NET Core only.";
```

Comment:
This is a general change with this PR: previously we had two sets of tests. Some tests tested "classic" projects and those tests ran in .NET Framework. Some tests tested SDK-style projects, and those ran in .NET Core. That was necessary because until this PR, that was the only supported combo. Since .NET Framework projects can now launch a .NET Core OOP process to read SDK projects and vice versa, this goes from unsupported to supported, and thus now in need of testing!